leandra-project/src/components/PullRequestStatus.js

```diff
@@ -16,24 +14,10 @@ const PullRequestStatus = (props)=> {
 
         return (
             <p className="flex-parent">
-                {props ? props.pullStatus(props.status) : ""}
```

Comment:
It's not legal to dispatch redux actions within the ***render body*** of the component function. 

An exception to this rule would be through the use of hooks. 

Initially, by dispatching a redux action in the render body of the component function (i.e. `<PullRequestStatus />`), the redux will update the redux store state accordingly. This update will propagate to connected components, such as `<PullRequestStatus />`, which, when rendered, will dispatch a redux action in its render body. Because of the one-way data flow between React and redux, an infinite "update/render loop" can occur. 

A solution to this would be to simply not dispatch an action in the render function/body of the component; and to find a different approach to achieve the same goal.